### PR TITLE
Attempt to add a new action to batch registration: REGISTER_FIELDS

### PR DESCRIPTION
**Pull request by Industrial**
Hi.

I have a big form with 206 inputs (https://github.com/erikras/redux-form/issues/2118#issuecomment-300196560). Currently it takes a lot of time to render the page. It's even worse (6 to 10 seconds) if you have the devtools open. This is largely due to the fields registering one by one.

This is my attempt at adding an action to batch the registering of actions on the initial load.

I am trying to move the registering of the fields from the `Field` component to the `Form` component by collecting all the fields at `Field#componentWillMount` and doing the register in `Form#componentDidMount`.

The page load is now 1.2 seconds for me, down from 7 with devtools performance analyzer.

All the tests are passing except for the FieldArray ones. I might need some help with this. Would you be interested in adding this functionality @erikras ?

gr,

Tom

### CODE REVIEW

File: src/createField.js [hunk 1]
```diff
@@ -29,6 +29,22 @@ const createField = ({deepEqual, getIn, setIn, toJS}) => {
...
+          'Field',
+          () => this.props.validate,
+          () => this.props.warn
+        )
+        return
+      }
+
+      if (registeredFields[this.name]) {
+        this.context._reduxForm.unregister(this.name)
+      }
```
Comment (erikras) [inline src/createField.js]: Why are we unregistering on `cWM`? Because it will be reregistered on `cDM`?

Comment (Industrial) [inline src/createField.js]: I'll remove it, since it has no function I think.

**Comment by Industrial** [comment]
@erikras: There's still some tests failing. I'm not sure how to fix FieldArray to make them pass.

**Comment by HankMcCoy** [comment]
FYI, I am working on a very similar PR here: https://github.com/HankMcCoy/redux-form/pull/1/commits/ad50754, which I hope to have in a reviewable state by the end of the week. Just wanted to double check that the idea of adding batched registrations/unregistrations as an opt-in is something that you'd be potentially be interested in merging @erikras?

**Comment by erikras** [comment]
Sorry for the delay. Yes, I would be interested in merging such functionality, @Industrial, @HankMcCoy.

**Comment by tyscorp** [comment]
Does this PR work in React 16 also?

**Comment by erikras** [comment]
Sadly, due mostly to neglect on my side, this PR is fairly out of date. It might be quicker to resubmit it on the current `master`.

**Comment by Industrial** [comment]
I have not used redux-forms (nor had a chance to) in quite some time.

I made the PR to explore the possibility of this functionality but I don't
think I will be developing it further.

I hope Hank McCoy will have an attempt

**Comment by HankMcCoy** [comment]
I'm hoping to get https://github.com/HankMcCoy/redux-form/commit/ad50754 polished into an actual PR, based off of 7.1.0 this weekend.

**Comment by bertho-zero** [comment]
Having a 'CHANGE_FIELDS' action would also be useful for components that manage multiple fields at once.

**Comment by erikras** [comment]
I'm closing this due to git-rot, but I'd be open to the same thing being implemented fresh against the latest codebase.

**Pull request closed without merging**